src/event/entity/EntityTeleportEvent.php

```diff
@@ -36,10 +36,10 @@ class EntityTeleportEvent extends EntityEvent implements Cancellable{
 	use CancellableTrait;
 
 	public const CAUSE_PLUGIN = 0;
-	public const CAUSE_ITEM = 1;
+	public const CAUSE_PROJECTILE = 1;
```

Comment:
CAUSE_PROJECTILE doesn't tell us what type of projectile was used. Maybe CAUSE_ENDER_PEARL would be better?

Reply:
You know, I think the cause should be a string so plugins can specify their custom causes like "more pearls:snowball". That way we can be specific while also allowing customization

Reply:
Yeah agree, with ints its really easy to mess up if someone was to create custom one